### PR TITLE
Subscription: Subscription Management On ConfigNode

### PR DESCRIPTION
**Pull request by MiniSho**
## Description
we provide the interfaces to support managing Subscriptions:
About Topic:
1.  `CREATE TOPIC <topicName> 
WITH (
  [<parameter> = <value>,],
)`
2. `DROP TOPIC <topicName> `
3. `SHOW TOPICS`
4. `SHOW TOPIC <topicName>`

About Subscription:
1. `SHOW SUBSCRIPTIONS`
2. `SHOW SUBSCRIPTIONS ON <topicName>`

## Contributor
@MiniSho self-reviewed:  ✅
@DanielWang2035  self-reviewed: ✅

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/subscription/agent/AbstractSubscriptionAgent.java
Comment (DanielWang2035) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/subscription/agent/AbstractSubscriptionAgent.java:1]: Do we need this agent on CN? Isn't it for metadata sync only?

Comment (MiniSho) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/subscription/agent/AbstractSubscriptionAgent.java:1]: nice catch, that's my mistake. I've already removed it

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/subscription/SubscriptionTableResp.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    final List<TShowSubscriptionInfo> showSubscriptionInfoList = new ArrayList<>();
+
+    for (SubscriptionMeta subscriptionMeta : allSubscriptionMeta) {
+      showSubscriptionInfoList.add(
+          new TShowSubscriptionInfo(
+              subscriptionMeta.getTopicName(),
+              subscriptionMeta.getConsumerGroupID(),
+              subscriptionMeta.getConsumerIDs()));
+    }
+    return new TShowSubscriptionResp(status);
```
Comment (VGalaxies) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/subscription/SubscriptionTableResp.java:68]: showSubscriptionInfoList unused?

**Review by VGalaxies — COMMENTED** [review]
some suggestions~

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/subscription/SubscriptionTableResp.java (hunk 1 above)
Comment (DanielWang2035) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/subscription/SubscriptionTableResp.java]: fixed

**Review by Caideyipi — COMMENTED** [review]
Add ser/de tests for newly added plans, procedures and metas.

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**